cli/cmd/healthcheck_extensions.go

```diff
@@ -0,0 +1,315 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/briandowns/spinner"
+	"github.com/linkerd/linkerd2/pkg/healthcheck"
+	"github.com/linkerd/linkerd2/pkg/version"
+	"github.com/mattn/go-isatty"
+	utilsexec "k8s.io/utils/exec"
+)
+
+// glob is satisfied by filepath.Glob.
+type glob func(string) ([]string, error)
+
+// extension contains the full path of an extension executable. If it's a
+// a built-in extension, the builtin field is populated with its name.
```

Comment:
probably worth clarifying that these fields are mutually exclusive: one or the other must be set but not both.

Reply:
both are set for `builtin`, it'll look like this:
```go
{
  path: "/home/sig/.linkerd2/bin/linkerd",
  builtin: "viz",
}
```